### ISSUE TITLE
Unhandled Exception: System.Exception: Unable to find crypto stream TypeDef #8

### ISSUE THREAD

**Issue by Masterxv** [HoLLy-HaCKeR/EazFixer#27]
Unhandled Exception: System.Exception: Unable to find crypto stream TypeDef

Unhandled Exception: System.Exception: Unable to find crypto stream TypeDef
at eazdevirt.EazModule.Initialize() in E:\xxxx\eazdevirt-master\src\eazdevirt\Core\EazModule.cs:line 86
at eazdevirt.EazModule..ctor(ModuleDefMD module, ILogger logger) in E:\xxx\eazdevirt-master\src\eazdevirt\Core\EazModule.cs:line 77
at eazdevirt.Program.TryLoadModule(String path, ILogger logger, EazModule& module) in E:\xxx\eazdevirt-master\src\eazdevirt\Program.cs:line 192
at eazdevirt.Program.DoDevirtualize(MonoOptions options) in E:\xxx\eazdevirt-master\src\eazdevirt\Program.Devirtualize.cs:line 168
at eazdevirt.Program.Main(String[] args) in E:\xxx\eazdevirt-master\src\eazdevirt\Program.cs:line 139

**Comment by HoLLy-HaCKeR** [comment]
Please upload your sample, or a sample that can reproduce the bug. Make sure your sample is actually obfuscated with EazFuscator.

**Comment by adstep** [comment]
I was seeing the same issue. [Here](https://beatmods.com/uploads/5e8238d403dce96ca6cf0dbd/universal/ScoreSaber-2.3.1.zip) is the dll that I'm working with that hits the issue.

Somewhere along the way Eazfuscator stopped having CryptoStream inherit from .NET Stream and rolled their own implementation of 'CustomStream' to inherit from. I was able to work around the issue by removing a constraint check in ```FindCryptoStream``` for the module type to inherit from a Stream. This usually still finds the correct Crypto stream type anyways.

In addition to the CustomStream, it also uses its own version of .NET BinaryReader. This does some additional operations on top of the encryption the CryptoStream does. I think this 'CustomBinaryReader' is version dependent as I can see some of the additional operations within the reader change based on different historical versions of the DLL I'm trying to analyze.

I've managed to replace most of the relevant bits that read the CryptoStream to use my CustomBinaryReader, but I'm still hitting hitting issues devirtualizing. It still seems to be reading garbage from the resource stream.

I've created a [fork](https://github.com/adstep/eazdevirt/tree/adstep/master/customStream) with my work. It is in no-way working. Its CustomBinaryReader implementation is tied to the dll I provided above. Hoping someone can find it useful.
